types/types_test.go

```diff
@@ -127,14 +127,19 @@ func TestMemMarker_Count(t *testing.T) {
 	require.Equal(t, 1, countByState(AlertStateActive))
 	require.Equal(t, 1, countTotal())
 
-	// Insert a suppressed alert.
+	// Insert a suppressed alert by silence.
 	marker.SetActiveOrSilenced(a2.Fingerprint(), 1, []string{"1"}, nil)
 	require.Equal(t, 1, countByState(AlertStateSuppressed))
 	require.Equal(t, 2, countTotal())
 
-	// Insert a resolved alert - it'll count as active.
+	// Insert a resolved silenced alert - it'll count as suppressed.
 	marker.SetActiveOrSilenced(a3.Fingerprint(), 1, []string{"1"}, nil)
-	require.Equal(t, 1, countByState(AlertStateActive))
+	require.Equal(t, 2, countByState(AlertStateSuppressed))
+	require.Equal(t, 3, countTotal())
+
+	// Insert a resolved alert - it'll count as active.
```

Comment:
I think this better explains what is happening here, what do you think?

```suggestion
	// Remove the silence for a3 - it'll count as active.
```